### DISCUSSION TITLE
Decide How to Improve Use with React Server Components

### DISCUSSION THREAD

**Discussion by tulup-conner** [themesberg/flowbite-react#940] [Polls]
Last week, we created [a discussion on how to improve the development experience for Next.js 13 users and Flowbite React](https://github.com/themesberg/flowbite-react/discussions/923).

We received a little feedback, and now we would like to open a simple poll where you can vote directly on the solutions proposed so far.

The outcome of this poll won't necessarily determine the action taken, but it will be a starting point. There may be further polls in the near future, especially if more ideas are contributed to the discussion.

## **Please check out [the discussion](https://github.com/themesberg/flowbite-react/discussions/923) to learn more details about the proposed solutions**. The solutions are not all created equal. Some would require substantial time investment to test, **and they may not even work for all users.**

**Comment by zoltanszogyenyi** [comment]
I think it's worth exploring the third and fourth options first and if they don't work or involve a large enough developer experience drop then we can fall back for the second option which would automatically make all Flowbite React components client components.

Creating and incorporating a custom build command could also be a good solution especially if we can do it in a way that doesn't hinder the developer experience (ie. changes are applied on file save via a watcher).

Though, it would be interesting to see other solution propositions as well!

**Comment by m-torin** [comment]
use client is a clear and easy way to add support. It’s also standard in many other React libraries. Additionally, it fixes the problems short term while other ideas can be debated and planned.

**Reply by LuisValgoi** [reply]
@m-torin how about the pages we rely on this lib that require SEO? I don't see how any of the solutions above fixes it. Maybe I am missing something... @tulup-conner any thoughts?

**Comment by rluders** [comment]
Folks,

Please test the [0.7.0-beta.2](https://www.npmjs.com/package/flowbite-react/v/0.7.0-beta.2) release. It introduces some RSC support and improves the theme by converting it from Context to Singleton.

It shouldn't introduce any breaking change. If you find any issues, please, report them as an issue, and don't forget to say that you are using the beta version.

**Comment by SutuSebastian** [comment]
Update: [0.7.0-beta.3](https://www.npmjs.com/package/flowbite-react/v/0.7.0-beta.3) is out.
